contracts/Example2.sol

```diff
@@ -4,8 +4,13 @@ contract Example2 {
     uint public counter;
  
     function incrementBy(uint[] memory arr) public {
+        uint total = counter;
         for (uint idx = 0; idx < arr.length; idx++){
-            counter += arr[idx];
+            assembly {
```

Comment:
You should only use assembly when it is not possible to implement it in solidity since it makes the code a lot more confusing to read